### PR TITLE
GDV-489: Add legend component for visible layers

### PR DESCRIPTION
**Pull request by IoannaMi**

### CODE REVIEW

File: src/components/DataLayersCard.vue [hunk 1]
```diff
@@ -174,15 +186,15 @@
           this.$router.push({ path, params })
         } else {
           this.$router.push('/data')
-        }
-
-        //find the layer of the dataset to load on the map based on the chosen values only if the dataset is visible
+        }        
         if (!dataset.visible) {
           this.clearActiveDatasetIds()
+          //TODO: check if there is a better way. I need the layerId to check if the layer is in the activeMapboxLayers
```
Comment (CindyvdVries) [inline src/components/DataLayersCard.vue:192]: as in using the params from the router?? (we do this in a few other places in the dgds-front-end code as well)
this is being done also in a few places in the coclico code I see, maybe make a central action in the store or something?

File: src/store/map/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import getCatalog from '@/lib/request/get-catalog'
 import datasets from './datasets.js'
 import buildGeojsonLayer from '@/lib/mapbox/build-geojson-layer'
+import matchLayerIdToProperties from '@/lib/match-layerId-to-properties.js'
```
Comment (CindyvdVries) [inline src/store/map/index.js:4]: layer-id (mixup camelCase and snake_case)

File: src/store/map/index.js [hunk 3]
```diff
@@ -103,6 +113,41 @@ export default {
...
+      
+      const newMin = _.get(dataset, 'properties.deltares:min', '')
+      const newMax =  _.get(dataset, 'properties.deltares:max', '')
+      const datasetId = _.get(dataset, 'id')
+      
+      
+      const mapboxLayer = state.activeMapboxLayers.find(({id}) => id.includes(datasetId))
+      const mapboxLayerId = _.get(mapboxLayer, 'id')
+      const circleColors = _.get(mapboxLayer, 'paint.circle-color')
+      //remove mapboxlayer in order to re-add it with the new colors
```
Comment (CindyvdVries) [inline src/store/map/index.js:127]: update paint of existing layer?

**Review by CindyvdVries — CHANGES_REQUESTED** [review]
The tests for the store should be updated and please run npm run lint -- --fix!

I don't see it working, is this dependent on a certain version of the coclico-data repo?

**Comment by IoannaMi** [comment]
> The tests for the store should be updated and please run npm run lint -- --fix!



> The tests for the store should be updated and please run npm run lint -- --fix!
> 
> I don't see it working, is this dependent on a certain version of the coclico-data repo?

Yep it is dependent on certain version indeed. Still waiting for the properties that contains the colors to be on the live version.

**Comment by IoannaMi** [comment]
> > The tests for the store should be updated and please run npm run lint -- --fix!
> 
> > The tests for the store should be updated and please run npm run lint -- --fix!
I started at the end of the previous sprint working on the tests but in another branch. Not in this one.
> > I don't see it working, is this dependent on a certain version of the coclico-data repo?
> 
> Yep it is dependent on certain version indeed. Still waiting for the properties that contains the colors to be on the live version. It works only locally at this moment if you change to this certion version of the coclico-data repo

**Pull request merged**